benchmarks/benchmarks/bench_lib.py

```diff
@@ -132,11 +132,26 @@ def setup(self, array_size, percent_nans):
         # produce a randomly shuffled array with the
         # approximate desired percentage np.nan content
         base_array = np.random.uniform(size=array_size)
-        base_array[base_array < percent_nans / 100.] = np.nan
+        n_nan = int(percent_nans * array_size)
+        nan_indices = np.random.randint(array_size, size=n_nan)
```

Comment:
This would scew at high percentages, which isn't a big deal, but maybe no point in it?  `np.random.choice` is probably way to get the correct number.

Reply:
Switched over to `np.random.choice` :)